### PR TITLE
Remove contract check to avoid crash

### PR DESCRIPTION
**Pull request by tmat**
Workaround for [1459993](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1459993)/[AB#1356990](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1356990).

Real fix would be https://github.com/dotnet/roslyn/issues/56815, which is currently blocked on missing Editor API.

### CODE REVIEW

**Comment by tmat** [comment]
@jinujoseph for approval

File: src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs [hunk 1]
```diff
@@ -220,9 +220,9 @@ private static void SetValueForOption(IOption option, ref LANGPREFERENCES3 langu
         public bool TryFetch(OptionKey optionKey, out object value)
         {
             // This particular serializer is a bit strange, since we have to initially read things out on the UI thread.
-            // Therefore, we refresh the values in the constructor, meaning that this should never get called for our values.
-
-            Contract.ThrowIfTrue(_supportedOptions.Contains(optionKey.Option) && _languageMap.ContainsKey(optionKey.Language));
+            // Therefore, we refresh the values in the constructor, meaning that this should never get called for our values,
+            // unless the constructor fails to fetch the values. We accept that case for now.
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs:224]: Why did the constructor fail?

Comment (tmat) [reply src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs:224]: IDK, but there is this check:

https://github.com/dotnet/roslyn/pull/58675/files#diff-2ba3ebb4519f577240311d8da163bf6e9d0908ea7563065ace41cc93992ed1e7R72

Comment (jasonmalinowski) [reply src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs:224]: Can we add a non-fatal fault report here then so we have a chance to root cause this? The comment there implies we're handling the case where the language isn't installed which seems fine...but if it's not installed then why are we asking for indentation/formatting options for it? That seems suspicious.

Comment (tmat) [reply src/VisualStudio/Core/Def/Implementation/Options/LanguageSettingsPersister.cs:224]: I don't know if we care about this. The whole LanguageSettingsPersister.cs will go away once we get https://github.com/dotnet/roslyn/issues/56815 implemented.

**Review by sharwell — CHANGES_REQUESTED** [review]
Marking Request Changes to give more time to review

**Review by sharwell — CHANGES_REQUESTED** [review]
I'd like to see one of the following alternatives instead:

1. Update the constructor to call `RefreshLanguageSettings` for all languages (including ones where GetUserPreferences4 fails). For the failure case, we can initialize settings with default values.
2. Update the contract check to not throw for cases where GetUserPreferences4 was called but failed

**Comment by tmat** [comment]
@sharwell Done: [2]

**Comment by tmat** [comment]
@jasonmalinowski Ping

**Review by jasonmalinowski — APPROVED** [review]
:shipit:

**Review by CyrusNajmabadi — APPROVED** [review]

**Pull request merged**